traces/stf_trace_gen/coremark_riscv.patch

```diff
@@ -0,0 +1,835 @@
+diff --git a/core_main.c b/core_main.c
+index a4beeb6..6b6a706 100644
+--- a/core_main.c
++++ b/core_main.c
+@@ -20,6 +20,7 @@ Original Author: Shay Gal-on
+         This file contains the framework to acquire a block of memory, seed
+    initial parameters, tun t he benchmark and report the results.
+ */
++#include "../trace_macros.h"
+ #include "coremark.h"
+ 
+ /* Function: iterate
+@@ -263,6 +264,7 @@ for (i = 0; i < MULTITHREAD; i++)
+     }
+     /* perform actual benchmark */
+     start_time();
++	START_TRACE
+ #if (MULTITHREAD > 1)
+     if (default_num_contexts > MULTITHREAD)
+     {
+@@ -281,6 +283,7 @@ for (i = 0; i < MULTITHREAD; i++)
+ #else
+     iterate(&results[0]);
+ #endif
++	STOP_TRACE
+     stop_time();
+     total_time = get_time();
+     /* get a function of the input to report */
+diff --git a/riscv64/core_portme.c b/riscv64/core_portme.c
+new file mode 100755
+index 0000000..8f17cb8
+--- /dev/null
++++ b/riscv64/core_portme.c
```

Comment:
Did you intend for this code to be in the patch?

Reply:
Yes, because coremark needs a few function definitions which are platform dependant.

Reply:
Ok, where did the definitions originate?  Although the licensing is the same as Olympia's I do worry that if the author _changes_ that license and we are have this separate patch that does not align with that, there will be trouble.  Can you provide directions on how to get this patch instead of including it?